src/PanelTraits/Create.php

```diff
@@ -120,4 +144,94 @@ public function syncPivot($model, $data, $form = 'create')
+    {
+        $relationData = $this->getRelationDataFromFormData($data, $form);
+        $this->createRelationsForItem($item, $relationData);
+    }
+
+    /**
+     * Create any existing one to one relations for the current model from the relation data.
+     *
+     * @param Model $item The current CRUD model.
+     * @param array $formattedData The form data.
+     */
+    private function createRelationsForItem($item, $formattedData)
+    {
+        foreach ($formattedData['relations'] as $relationMethod => $relationData) {
+            $model = $relationData['model'];
+            $relation = $item->{$relationMethod}();
+
+            if ($relation instanceof BelongsTo) {
+                $modelInstance = $model::find($relationData['values'])->first();
+                if ($modelInstance != null) {
+                    $relation->associate($modelInstance)->save();
+                } else {
+                    $relation->dissociate()->save();
+                }
+            } elseif ($relation instanceof HasOne) {
+                if ($item->{$relationMethod} != null) {
+                    $item->{$relationMethod}->update($relationData['values']);
+                    $modelInstance = $item->{$relationMethod};
+                } else {
+                    $relationModel = new $model();
+                    $modelInstance = $relationModel->create($relationData['values']);
+                    $relation->save($modelInstance);
+                }
+            } else {
+                $relationModel = new $model();
+                $modelInstance = $relationModel->create($relationData['values']);
+                $relation->save($modelInstance);
+            }
+
+            if (isset($relationData['relations'])) {
+                $this->createRelationsForItem($modelInstance, ['relations' => $relationData['relations']]);
+            }
+        }
+    }
+
+    /**
+     * Get a relation data array from the form data. For each relation defined in the fields through the entity
+     * attribute, set the model, the parent model and the attribute values. For relations defined with the "dot"
+     * notations, this will be used to calculate the depth in the final array (@see \Illuminate\Support\Arr::set() for more).
+     *
+     * @param array $data The form data.
+     * @param string $form Optional form type. Can be either 'create', 'update' or 'both'. Default is 'create'.
+     *
+     * @return array The formatted relation data.
+     */
+    private function getRelationDataFromFormData($data, $form = 'create')
+    {
+        $relationFields = $this->getRelationFields($form);
+
+        $relationData = [];
+        foreach ($relationFields as $relationField) {
+            if (array_key_exists($relationField['name'], $data) && empty($relationField['pivot'])) {
+                $key = implode('.relations.', explode('.', $relationField['entity']));
+                $fieldData = array_get($relationData, 'relations.'.$key, []);
+
```

Comment:
Huh?

Let's pretend I'm the Pointy Haired Boss (i.e. @tabacitu's evil twin).

If I told you, "Please lift the above two lines into a private function with a sensible name" what would the sensible name be?

Reply:
This whole method takes the form data and creates a new relation data array. 


Since each relation name from the relation string can be one of one-to-one, one-to-many, many-to-many, I decided to use the following format for the array:

```php
[
	'relations' => [
		'relationName1' =>[
				'parent' => '<fully qualified name of the model in which the relation has been defined>'
				'model' => '<fully qualified name of the model resulting from the relation defined in the parent>'
				'values' => [
					'<array with all the form values defined for this relation>'
				]
				'relations' => [
					'subRelationName1' => [...]
					'subRelationName2' => [...]
					...
				]
		]
		'relationName2' =>[...]
		...
	]
]
```

Basically, the implode and explode line concatenates a 'relations' keyword in the relation string (defined in `$relationField['entity']`), so I can then use `array_get` and `array_set` on the relation array to easily set and retrieve data using the dot notation.

For a relation string like 'details.categories', this would turn it into relations.details.relations.categories'.

Reply:
Errr, but in that example very second one is `relations` - is it possible that there would be a `relations.details.foobars.bingboes.relations.categories.automobiles.relations.bobos`?